File: example/spring/provider/src/main/java/poppet/example/spring/service/UserInternalService.java
package poppet.example.spring.service;

import org.springframework.stereotype.Service;
import poppet.example.spring.model.User;

@Service
public class UserInternalService implements UserService {
    @Override
    public User findById(String id) {
        return new User(id, "Antony");
    }
}

File: example/play/provider/app/poppet/example/play/service/UserInternalService.scala
package poppet.example.play.service

import poppet.example.play.model.User
import poppet.example.play.service.UserService
import scala.concurrent.Future

class UserInternalService extends UserService {
    override def findById(id: String): Future[User] = {
        //emulation of business logic
        if (id == "1") Future.successful(User(id, "Antony"))
        else Future.failed(new IllegalArgumentException("User is not found"))
    }
}
